src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,65 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE;
+
+    private T[] elementData;
+    private int size;
+
+    public MateStack() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     public void push(T value) {
+        if (value == null) {
+            return;
+        }
+
+        resize();
+        elementData[size] = value;
+        size++;
+    }
+
+    private void resize() {
+        if (size == elementData.length) {
+            int newCapacity = elementData.length * 2;
+            if (newCapacity - MAX_ARRAY_SIZE > 0) {
+                newCapacity = MAX_ARRAY_SIZE;
+            }
 
+            T[] newArray = (T[]) new Object[newCapacity];
+            System.arraycopy(elementData, 0, newArray, 0, elementData.length);
+            elementData = newArray;
+        }
     }
```

Comment:
after changes, this method can be simplified to something like this (Creating constant for default multiplayer is good approach in this case):

```java
private void resize() {
        currentSize = (int) (currentSize * DEFAULT_MULTIPLAYER);
        System.arraycopy(stack,0,stack,0, currentSize);
    }
```